hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/capacity/CapacitySchedulerQueueCapacityHandler.java

```diff
@@ -70,61 +70,133 @@ public CapacitySchedulerQueueCapacityHandler(RMNodeLabelsManager labelsManager)
+    QueueCapacityUpdateContext updateContext =
+        new QueueCapacityUpdateContext(clusterResource, labelsManager);
+
+    RootCalculationDriver rootCalculationDriver = new RootCalculationDriver(rootQueue, updateContext,
         rootCalculator, definedResources);
     rootCalculationDriver.calculateResources();
-    queue.refreshAfterResourceCalculation(updateContext.getUpdatedClusterResource(), resourceLimits);
+    rootQueue.refreshAfterResourceCalculation(updateContext.getUpdatedClusterResource(), resourceLimits);
   }
 
-  private void updateChildren(
-      CSQueue parent, QueueCapacityUpdateContext updateContext,
-      ResourceLimits resourceLimits) {
-    if (parent == null || CollectionUtils.isEmpty(parent.getChildQueues())) {
+  private void update(
+      CSQueue queue, QueueCapacityUpdateContext updateContext, ResourceLimits resourceLimits) {
+    if (queue == null || CollectionUtils.isEmpty(queue.getChildQueues())) {
       return;
     }
 
     ResourceCalculationDriver resourceCalculationDriver = new ResourceCalculationDriver(
-        parent, updateContext, calculators, definedResources);
+        queue, updateContext, calculators, definedResources);
     resourceCalculationDriver.calculateResources();
 
     updateChildrenAfterCalculation(resourceCalculationDriver, resourceLimits);
   }
 
   private void updateChildrenAfterCalculation(
       ResourceCalculationDriver resourceCalculationDriver, ResourceLimits resourceLimits) {
-    for (CSQueue childQueue : resourceCalculationDriver.getParent().getChildQueues()) {
-      resourceCalculationDriver.setCurrentChild(childQueue);
-      resourceCalculationDriver.updateChildCapacities();
-      ResourceLimits childLimit = ((ParentQueue) resourceCalculationDriver.getParent()).getResourceLimitsOfChild(
-          childQueue, resourceCalculationDriver.getUpdateContext().getUpdatedClusterResource(), resourceLimits, NO_LABEL, false);
+    ParentQueue parentQueue = (ParentQueue) resourceCalculationDriver.getQueue();
+    for (CSQueue childQueue : parentQueue.getChildQueues()) {
+      updateChildCapacities(resourceCalculationDriver, childQueue);
+
+      ResourceLimits childLimit = parentQueue.getResourceLimitsOfChild(childQueue,
+          resourceCalculationDriver.getUpdateContext().getUpdatedClusterResource(),
+          resourceLimits, NO_LABEL, false);
       childQueue.refreshAfterResourceCalculation(resourceCalculationDriver.getUpdateContext()
               .getUpdatedClusterResource(), childLimit);
-      updateChildren(childQueue, resourceCalculationDriver.getUpdateContext(), childLimit);
+
+      update(childQueue, resourceCalculationDriver.getUpdateContext(), childLimit);
+    }
+  }
+
+  /**
+   * Updates the capacity values of the currently evaluated child.
+   * @param childQueue child queue on which the capacities are set
+   */
+  private void updateChildCapacities(ResourceCalculationDriver resourceCalculationDriver, CSQueue childQueue) {
```

Comment:
This is a bit misleading again, this is just simply updateQueueCapacities, there is nothing child specific thing involved here. So we can simply go with the queue terminology. Child should be always relative to something, but if the only thing we are working with is the child, then it's not a child but a queue.

Reply:
It is the same here as well: we named the parent queue on which the handler was invoked as queue, then passing an other queue as parameter is ambiguous. This child is relative to the queue, which is stored in the driver. Since we always do calculations in batch under a specific parent, we need to distinguish the difference between the children we are actually using in the calculators and their common parent.